bot/cogs/queue.py

```diff
@@ -200,3 +212,100 @@ async def cap_error(self, ctx, error):
             missing_perm = error.missing_perms[0].replace('_', ' ')
             embed = self.bot.embed_template(title=f'Cannot change queue capacity without {missing_perm} permission!')
             await ctx.send(embed=embed)
+
+    async def is_banned(self, guild, user):
+        """ Check if a user is banned from queueing and remove any ende bans for their guild. """
+        bans = await self.bot.db_helper.get_banned_users(guild.id)
+        now = datetime.now(timezone.utc)
+        ended_bans = [user_id for user_id, unban_time in bans.items() if unban_time is not None and now > unban_time]
+        await self.bot.db_helper.delete_banned_users(guild.id, ended_bans)
```

Comment:
It seems a little silly to do this via Python when it's possible to do this in a query. (it'd also be faster to do it in the query)

Reply:
I was looking up the best way to do this in PostgreSQL and found someone saying it would be better to do this outside of the DB because it would be too performance intensive. I was hesitant about it too. I'm going to check again to see if I can find a solution that works. It would be much more fitting for it to function inside of the DB.